components/admin_console/server_logs/logs.tsx

```diff
@@ -62,15 +54,14 @@ export default class Logs extends React.PureComponent {
 
     reload = () => {
         this.setState({loadingLogs: true});
-        this.props.actions.getLogs(this.state.page, this.state.perPage).then(
-            () => this.setState({loadingLogs: false}),
-        );
+        this.props.actions.getLogs(this.state.page, this.state.perPage);
+        this.setState({loadingLogs: false});
     }
 
     render() {
         let content = null;
 
-        if (this.state.loadingLogs) {
+        if (this.state.loadingLogs || !this.props.logs || this.props.logs.length === 0) {
```

Comment:
Instead of setting `this.state.loadingLogs` in `componentDidUpdate` and `componentDidMount`, I move such a check indirectly here in `render()`.

We should not use `this.setState` in `componentDidUpdate` and `componentDidMount` moving forward. If anyone has better idea, please comment on this PR. :)

Reply:
It's usually good to avoid `setState` in those methods, but this does slightly change the behaviour here since it:
1. Removes the loading indicator when switching pages
2. Shows the loading indicator if there were somehow no logs visible

We might want to switch back to setting `loadingLogs` after the calls to `getLogs` in those methods to fix those. While it's not good to call `setState` immediately in either of those, it's okay to do it after an async call to reflect those changes. The main reason you usually don't want to call it outside of an async call is to avoid rendering the component twice in a row (or more!) since that can cause problems.

Reply:
Actually, never mind all of that. Looking at the previous code, `loadingLogs` is actually never set back to `true` when changing pages, so this isn't a functional change anyway.

No changes are needed here then since there's no way that the logs are empty given we print to them immediately as the server starts up.